### PR TITLE
Add org.zaproxy.ZAP

### PR DESCRIPTION
**Pull request by kurobeats**
Honestly, I sat down today with no knowledge how to make a flatpak and here we are. It's functional.

### CODE REVIEW

**Comment by kurobeats** [comment]
bot, build org.zaproxy.ZAP

**Comment by kurobeats** [comment]
bot, build org.zaproxy.ZAP

**Comment by kurobeats** [comment]
bot, build org.zaproxy.ZAP

File: org.zaproxy.ZAP.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<component type="desktop">
```
Comment (bilelmoussaoui) [inline org.zaproxy.ZAP.appdata.xml]:
```suggestion
<component type="desktop-application">
```

File: org.zaproxy.ZAP.desktop [hunk 2]
```diff
@@ -0,0 +1,11 @@
+[Desktop Entry]
+Categories=System;Security;
+Type=Application
+Terminal=false
+Name=OWASP ZAP
+GenericName=The OWASP Zed Attack Proxy
+Comment=Automatically find security vulnerabilities in your web applications while you are developing and testing.
+Exec=zap
+Icon=org.zaproxy.ZAP.png
```
Comment (bilelmoussaoui) [inline org.zaproxy.ZAP.desktop]:
```suggestion
Icon=org.zaproxy.ZAP
```

**Comment by bilelmoussaoui** [comment]
The appdata and desktop file should be upstreamed. Anyone from upstream is interested in having write-access to the repository?

File: org.zaproxy.ZAP.json [hunk 3]
```diff
@@ -0,0 +1,59 @@
+{
+    "id": "org.zaproxy.ZAP",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "18.08",
```
Comment (LeandroStanger) [inline org.zaproxy.ZAP.json]: Update runtime to 19.08

Comment (bilelmoussaoui) [reply org.zaproxy.ZAP.json]: Not possible for now. Openjsdk still doesn't have a 19.08 branch.

**Comment by kurobeats** [comment]
> 
> 
> The appdata and desktop file should be upstreamed. Anyone from upstream is interested in having write-access to the repository?
They are tracking this in https://github.com/zaproxy/zaproxy/issues/4643 so its entirely possible they will at some stage.

**Comment by kurobeats** [comment]
bot, build org.zaproxy.ZAP

File: org.zaproxy.ZAP.json [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+                "install -Dm644 org.zaproxy.ZAP.appdata.xml /app/share/appdata/org.zaproxy.ZAP.appdata.xml",
+                "mkdir /app/share/owasp-zap",
+                "rm org.zaproxy.ZAP.png org.zaproxy.ZAP.desktop org.zaproxy.ZAP.appdata.xml",
+                "cp -r ./* /app/share/owasp-zap",
+                "ln -s /app/share/owasp-zap/zap.sh /app/bin/zap"
+            ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://github.com/zaproxy/zaproxy/releases/download/v2.8.0/ZAP_2.8.0_Linux.tar.gz",
```
Comment (thc202) [inline org.zaproxy.ZAP.json]: It could use 2.8.1 (it's the latest for Linux).

**Comment by kurobeats** [comment]
bot, build org.zaproxy.ZAP

File: org.zaproxy.ZAP.json (hunk 4 above)
Comment (kurobeats) [reply org.zaproxy.ZAP.json]: consider it done.

Comment (thc202) [reply org.zaproxy.ZAP.json]: Thanks :)

**Comment by kurobeats** [comment]
bot, build org.zaproxy.ZAP

**Comment by psiinon** [comment]
As per https://github.com/zaproxy/zaproxy/issues/4643 the ZAP core team are completely on-board with this and very happy for @kurobeats to lead this effort.

**Comment by psiinon** [comment]
The ZAP core team accounts to also add to the new repo are: @psiinon @thc202 and @kingthorin

File: org.zaproxy.ZAP.appdata.xml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    <p>
+      The OWASP Zed Attack Proxy (ZAP) is one of the world’s most popular 
+      free security tools and is actively maintained by hundreds of international 
+      volunteers. It can help you automatically find security vulnerabilities 
+      in your web applications while you are developing and testing your 
+      applications. Its also a great tool for experienced pentesters to use for 
+      manual security testing.
+    </p>
+    <p>
+      NOTE: This wrapper is not verified by, affiliated with, or supported by
```
Comment (barthalion) [inline org.zaproxy.ZAP.appdata.xml]: No need for such disclaimer given it's open source and upstream team is happy to have it on Flathub.

File: org.zaproxy.ZAP.json [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+                    "type": "file",
+                    "path": "org.zaproxy.ZAP.png"
+                },
+                {
+                    "type": "file",
+                    "path": "org.zaproxy.ZAP.desktop"
+                },
+                {
+                    "type": "file",
+                    "path": "org.zaproxy.ZAP.appdata.xml"
```
Comment (barthalion) [inline org.zaproxy.ZAP.json]: Ideally these files should be maintained by upstream team. (FYI @psiinon)

File: org.zaproxy.ZAP.json [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    "separate-locales": false,
+    "sdk-extensions": [
+        "org.freedesktop.Sdk.Extension.openjdk11" 
+    ],
+    "finish-args": [
+        "--share=ipc",
+        "--socket=x11",
+        "--share=network",
+        "--env=JAVA_HOME=/app/jre",
+        "--filesystem=home"
```
Comment (barthalion) [inline org.zaproxy.ZAP.json]: Does it need access to entire home or could be limited?

File: org.zaproxy.ZAP.appdata.xml (hunk 5 above)
Comment (kurobeats) [reply org.zaproxy.ZAP.appdata.xml]: sweet.

File: org.zaproxy.ZAP.json (hunk 6 above)
Comment (kurobeats) [reply org.zaproxy.ZAP.json]: I believe the guys are happy for this to be the case, but they (and I) will need direction on how to maintain upstream. Longer conversation for another time, maybe?

File: org.zaproxy.ZAP.json (hunk 7 above)
Comment (kurobeats) [reply org.zaproxy.ZAP.json]: Great question, you are right to point this out. There does need to be some interaction with the home directory (to save sessions, upload target lists, etc) but it doesn't need much more that Downloads and Documents for those purposes. I've amended the manifest.

**Comment by kurobeats** [comment]
bot, build org.zaproxy.ZAP

File: org.zaproxy.ZAP.json (hunk 6 above)
Comment (barthalion) [reply org.zaproxy.ZAP.json]: Sounds good, just please follow up on that. :+1:

**Comment by barthalion** [comment]
Repository has been created: https://github.com/flathub/org.zaproxy.ZAP


Thanks!

**Comment by barthalion** [comment]
@kingthorin, @psiinon, @thc202, @kurobeats, please make sure to accept invitation to collaborators to get actual write access to the reposiotry.

**Comment by psiinon** [comment]
Just accepted - many thanks @kurobeats and @barthalion 
So .. I'm now going to show my ignorance of flathub :/ 
Will ZAP now appear on https://flathub.org/ in the near future or does something else need to be done? I couldnt find it via https://flathub.org/apps/search/zap or https://flathub.org/apps/search/owasp
Just want to make sure that we publicize this as soon as we can :)

File: org.zaproxy.ZAP.json (hunk 7 above)
Comment (thc202) [reply org.zaproxy.ZAP.json]: ZAP creates a directory in the user's home (for main releases called `.ZAP`, which has the add-ons and other configuration files) would that still work with the new settings?

**Comment by thc202** [comment]
@barthalion could the wiki be disabled in the new repo?

**Comment by bilelmoussaoui** [comment]
Once a MR is pushed to master (for stable releases) or beta (for beta releases), the application is built and you can follow the progress here https://flathub.org/builds/. If the build was successful, there's a delay before the application got published to the store; you can either log in with your Github account, click on Zap build and click on publish or just wait until it's done automatically.

**Comment by barthalion** [comment]
@psiinon Pretty much wat Bilel wrote. At the time you wrote your comment it hasn't been published yet, but it is now: https://flathub.org/apps/details/org.zaproxy.ZAP

@thc202: Sure thing, disabled.

File: org.zaproxy.ZAP.json (hunk 7 above)
Comment (barthalion) [reply org.zaproxy.ZAP.json]: After `home` was replaced with `xdg-document` and `xdg-download`, no. You need to add `--persist=.ZAP` and flatpak will handle binding appropriate location.

In general it's good practice to limit filesystem access. Users can always run `flatpak override --filesystem=home org.zaproxy.ZAP` (or `host`) on their own.

**Pull request closed without merging**